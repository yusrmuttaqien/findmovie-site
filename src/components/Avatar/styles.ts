import styled from 'styled-components';

export const Wrapper = styled.figure`
  --_dimensions: 2.5rem;
  background-color: red;
  height: var(--_dimensions);
  width: var(--_dimensions);
  border-radius: 50%;
  overflow: hidden;

  img {
    height: var(--_dimensions);
    width: var(--_dimensions);
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    --_dimensions: 4rem;
  }
`;
