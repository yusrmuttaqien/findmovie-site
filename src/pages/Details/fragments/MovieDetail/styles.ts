import styled from 'styled-components';

export const Wrapper = styled.section`
  display: flex;
  flex-direction: column;
  gap: 1.5625rem;
  padding: 0 ${({ theme }) => theme.space.x.mobile};

  #metadatas {
    h3 {
      font-weight: 500;
      font-size: 1rem;

      &:not([data-meta='title']) {
        margin-bottom: 0.5rem;
      }

      span {
        font-weight: 800;
      }
    }

    > div {
      display: flex;
      flex-wrap: wrap;
      gap: 0.375rem;
    }

    &:last-child p {
      font-size: 0.85rem;
      font-weight: 200;
    }
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    gap: 2rem;
    padding: 0 ${({ theme }) => theme.space.x.desktop};

    #metadatas {
      h3 {
        font-size: 2.125rem;

        &:not([data-meta='title']) {
          margin-bottom: 1rem;
        }
      }

      > div {
        gap: 0.625rem;
      }

      &:last-child p {
        font-size: 1.5rem;
      }
    }
  }
`;
