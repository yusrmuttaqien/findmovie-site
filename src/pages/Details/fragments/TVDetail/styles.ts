import styled from 'styled-components';
import CC from 'components/ContentCard';

export const Wrapper = styled.section`
  display: flex;
  flex-direction: column;
  gap: 1.5625rem;
  padding: 0 ${({ theme }) => theme.space.x.mobile};

  #metadatas {
    h3 {
      font-weight: 500;
      font-size: 1rem;

      span {
        font-weight: 800;
      }
    }

    > div {
      margin-top: 0.5rem;
      display: flex;
      flex-wrap: wrap;
      gap: 0.375rem;
    }

    &:last-child p {
      margin-top: 0.5rem;
      font-size: 0.85rem;
      font-weight: 200;
    }
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    gap: 2rem;
    padding: 0 ${({ theme }) => theme.space.x.desktop};

    #metadatas {
      h3 {
        font-size: 2.125rem;
      }

      > div {
        margin-top: 1rem;
        gap: 0.625rem;
      }

      &:last-child p {
        margin-top: 1rem;
        font-size: 1.5rem;
      }
    }
  }
`;

export const ContentCard = styled(CC)`
  width: 100%;

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    width: calc(50% - ${({ theme }) => theme.space.x.desktop});
  }
`;
