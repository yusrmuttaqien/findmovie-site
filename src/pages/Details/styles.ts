import styled from 'styled-components';
import { Link as L } from 'react-router-dom';
import { Read as R } from 'assets/svgs';
import NIDetails from 'assets/nativeSVGs/NoImageDetails.svg?url';

export const Wrapper = styled.main`
  --_iconHeight: 7.4375rem;

  header {
    position: relative;
    padding: 0 ${({ theme }) => theme.space.x.mobile} calc(var(--_iconHeight) / 2 + 1rem);
    margin-bottom: calc(var(--_iconHeight) / 2 + 1rem);
    display: flex;
    flex-direction: column;
    justify-content: end;
    gap: 2rem;
    height: calc(100vh - var(--_iconHeight) / 2 - 1rem);
    min-height: calc(var(--_iconHeight) * 3 + 1rem);
    background: url(${NIDetails}) ${({ theme }) => theme.colors.white[100]};
    background-position: top center;
    background-repeat: no-repeat;

    #title-wrapper {
      h1,
      p {
        position: relative;
      }

      h1 {
        font-size: 4.1875rem;
        line-height: 92%;
        word-wrap: break-word;
        z-index: 1;
      }

      p {
        font-size: 1rem;
        z-index: 2;

        span {
          transition: color 0.1s ease-in-out;
        }
      }
    }
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    --_iconHeight: 17.5rem;

    header {
      padding: 0 ${({ theme }) => theme.space.x.desktop} calc(var(--_iconHeight) / 2 + 1rem);
      flex-direction: row;
      min-height: calc(var(--_iconHeight) * 1.5 + 1rem);

      #title-wrapper {
        height: max-content;
        align-self: flex-end;

        h1 {
          font-size: 11.15rem;
        }

        p {
          font-size: 1.75rem;
        }
      }
    }
  }
`;

export const Read = styled(R)`
  background-color: ${({ theme }) => theme.colors.black[30]};
  height: var(--_iconHeight);
  position: absolute;
  bottom: calc(-1 * var(--_iconHeight) / 2);
  left: 50%;
  transform: translateX(-50%);
  cursor: pointer;
  border-radius: 45%;
  backdrop-filter: blur(5px);
`;

export const Link = styled(L)`
  display: inline-flex;
  align-items: center;
  color: ${({ theme }) => theme.colors.white[50]};
  text-decoration: none;
  transition: color 0.1s ease-in-out;

  &:hover {
    color: ${({ theme }) => theme.colors.white[100]};

    + span {
      color: ${({ theme }) => theme.colors.white[50]};
    }
  }
`;

export const Backdrop = styled.div`
  position: absolute;
  inset: 0;

  &::after {
    content: '';
    position: absolute;
    inset: 0;
    background-color: ${({ theme }) => theme.colors.black[50]};
    backdrop-filter: blur(1px);
  }

  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
`;
