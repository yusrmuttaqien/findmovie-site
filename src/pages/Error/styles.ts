import styled from 'styled-components';

export const Wrapper = styled.main`
  display: grid;
  height: 100vh;
  place-items: center;
  font-weight: 600;

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    font-size: 2rem;
  }
`;
