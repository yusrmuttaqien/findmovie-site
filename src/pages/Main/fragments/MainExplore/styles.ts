import styled from 'styled-components';
import L from 'components/Loading';

export const Wrapper = styled.section`
  display: flex;
  flex-direction: column;
  padding: 0 ${({ theme }) => theme.space.x.mobile};
  gap: 2.25rem;

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    padding: 0 ${({ theme }) => theme.space.x.desktop};
    gap: 4.625rem;
  }
`;

export const Group = styled.div`
  display: flex;
  flex-direction: column;
  gap: 0.9375rem;

  h2 {
    font-weight: 600;
    font-size: 2rem;
  }

  #categories-wrapper {
    display: inherit;
    flex-direction: inherit;
    gap: 0.625rem;

    #categories {
      position: relative;

      h3 {
        font-weight: 500;
        font-size: 1rem;
        margin-bottom: 0.625rem;
        position: sticky;
        top: 0;
        left: 0;
        z-index: 2;
        backdrop-filter: blur(5px);
        background-color: ${({ theme }) => theme.colors.black[80]};
      }

      > div {
        position: relative;
        display: flex;
        flex-direction: column;
        gap: 0.625rem;
        z-index: 1;
      }
    }
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    gap: 2rem;

    h2 {
      font-size: 4rem;
    }

    #categories-wrapper {
      flex-direction: row;
      gap: 1.25rem;

      #categories {
        flex: 1;

        h3 {
          font-size: 2.125rem;
          margin-bottom: 2.125rem;
        }

        &:nth-child(2) h3 {
          text-align: right;
        }

        > div {
          gap: 1rem;
        }
      }
    }
  }
`;

export const Loading = styled(L)`
  height: 10rem;
`;
