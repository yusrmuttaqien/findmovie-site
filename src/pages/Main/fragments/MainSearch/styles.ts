import styled from 'styled-components';
import P from 'components/Pagination';

export const Wrapper = styled.section`
  padding: 0 ${({ theme }) => theme.space.x.mobile};

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    padding: 0 ${({ theme }) => theme.space.x.desktop};
  }
`;

export const Group = styled.div`
  display: flex;
  flex-direction: column;
  gap: 0.9375rem;

  #search-title {
    display: flex;
    align-items: baseline;

    p {
      font-size: 0.85rem;
      font-weight: 200;
    }

    h3 {
      font-weight: 500;
      font-size: 1rem;
    }
  }

  #categories-wrapper {
    display: inherit;
    flex-direction: inherit;
    gap: 0.625rem;

    #categories {
      display: inherit;
      flex-direction: inherit;
      position: relative;
      gap: inherit;

      h3 {
        font-weight: 500;
        font-size: 1rem;
        position: sticky;
        top: 0;
        left: 0;
        z-index: 2;
        backdrop-filter: blur(5px);
        background-color: ${({ theme }) => theme.colors.black[80]};
      }

      > div {
        display: inherit;
        flex-direction: column;
        gap: 0.625rem;
        z-index: 1;
      }
    }
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    gap: 2rem;

    #search-title {
      p {
        font-size: 2rem;
      }

      h3 {
        font-size: 2.125rem;
      }
    }

    #categories-wrapper {
      gap: 1.25rem;

      #categories {
        h3 {
          font-size: 2.125rem;
        }

        > div {
          display: grid;
          grid-template-columns: repeat(2, 1fr);
          grid-gap: 1.25rem;
        }
      }
    }
  }
`;

export const Pagination = styled(P)`
  position: sticky;
  top: 0;
  z-index: 2;
  backdrop-filter: blur(5px);
  background-color: ${({ theme }) => theme.colors.black[80]};
  margin-top: 1rem;

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    margin-top: 1.25rem;
  }
`;
