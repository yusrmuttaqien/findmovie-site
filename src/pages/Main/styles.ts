import styled from 'styled-components';
import SB from 'components/SearchBox';
import { Discover as D } from 'assets/svgs';

export const Wrapper = styled.main`
  --_iconHeight: 7.4375rem;

  header {
    position: relative;
    padding: 0 ${({ theme }) => theme.space.x.mobile} calc(var(--_iconHeight) / 2 + 1rem);
    margin-bottom: calc(var(--_iconHeight) / 2 + 1rem);
    display: flex;
    flex-direction: column;
    justify-content: end;
    gap: 2rem;
    border-bottom: 1px solid;
    border-color: ${({ theme }) => theme.colors.white[50]};
    height: calc(100vh - var(--_iconHeight) / 2 - 1.25rem);
    min-height: calc(var(--_iconHeight) * 2 + 1.25rem);

    #title-wrapper {
      h1 {
        font-size: 4.1875rem;
        line-height: 92%;
        word-spacing: 100vw;
      }

      p {
        font-size: 1rem;
      }
    }
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    --_iconHeight: 17.5rem;

    header {
      padding: 0 ${({ theme }) => theme.space.x.desktop} calc(var(--_iconHeight) / 2 + 1rem);
      flex-direction: row;
      min-height: calc(var(--_iconHeight) * 1.5 + 1.25rem);

      #title-wrapper {
        height: max-content;
        align-self: flex-end;

        h1 {
          font-size: 11.15rem;
        }

        p {
          font-size: 1.75rem;
        }
      }
    }
  }
`;

export const SearchBox = styled(SB)`
  align-self: center;

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    align-self: end;
  }
`;

export const Discover = styled(D)`
  background-color: ${({ theme }) => theme.colors.black[100]};
  height: var(--_iconHeight);
  position: absolute;
  bottom: calc(-1 * var(--_iconHeight) / 2);
  left: 50%;
  transform: translateX(-50%);
  cursor: pointer;

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    left: unset;
    right: ${({ theme }) => theme.space.x.desktop};
    transform: none;
  }
`;
