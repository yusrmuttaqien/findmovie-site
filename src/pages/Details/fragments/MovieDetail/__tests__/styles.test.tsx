import { describe, test, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Wrapper } from '../styles';
import theme from 'styles/index';
import 'jest-styled-components';

describe('Details page fragment: MovieDetail styles', () => {
  test('styled components should match snapshot', () => {
    const { asFragment: fWrapper } = render(<Wrapper theme={theme} />);

    expect(fWrapper()).toMatchSnapshot();
  });
});
