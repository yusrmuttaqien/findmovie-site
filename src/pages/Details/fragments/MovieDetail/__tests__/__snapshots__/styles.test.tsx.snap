// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Details page fragment: MovieDetail styles > styled components should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: column;
  gap: 1.5625rem;
  padding: 0 1.875rem;
}

.c0 #metadatas h3 {
  font-weight: 500;
  font-size: 1rem;
}

.c0 #metadatas h3:not([data-meta='title']) {
  margin-bottom: 0.5rem;
}

.c0 #metadatas h3 span {
  font-weight: 800;
}

.c0 #metadatas >div {
  display: flex;
  flex-wrap: wrap;
  gap: 0.375rem;
}

.c0 #metadatas:last-child p {
  font-size: 0.85rem;
  font-weight: 200;
}

@media screen and (min-width: 1279px) {
  .c0 {
    gap: 2rem;
    padding: 0 5.5rem;
  }

  .c0 #metadatas h3 {
    font-size: 2.125rem;
  }

  .c0 #metadatas h3:not([data-meta='title']) {
    margin-bottom: 1rem;
  }

  .c0 #metadatas >div {
    gap: 0.625rem;
  }

  .c0 #metadatas:last-child p {
    font-size: 1.5rem;
  }
}

<section
    class="c0"
  />
</DocumentFragment>
`;
